Tests/SonarScanner.MSBuild.Common.Test/RegexExtensionsTest.cs

```diff
@@ -0,0 +1,98 @@
+﻿/*
+* SonarAnalyzer for .NET
+* Copyright (C) 2015-2024 SonarSource SA
+* mailto: contact AT sonarsource DOT com
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+using System;
+using System.Text.RegularExpressions;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarScanner.MSBuild.Common;
+
+namespace SonarAnalyzer.Test.Extensions;
+
+[TestClass]
+public class RegexExtensionsTest
```

Comment:
I touched a regex and that triggered `T0004 Use 'SafeRegex.Matches' instead.` I therefore added the class from dotnet and the tests because it is needed on the long run anyway.